packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,249 @@
+      return 'List<' + rel.childSObject + '> ' + rel.relationshipName;
+    } else {
+      // expect the name to end with Id, then strip off Id
+      if (rel.field.endsWith('Id')) {
+        return (
+          rel.childSObject + ' ' + rel.field.slice(0, rel.field.length - 2)
+        );
+      } else {
+        return '';
+      }
+    }
+  }
+
+  private generateField(field: Field): string[] {
+    const decls: string[] = [];
+    if (field.referenceTo.length === 0) {
+      decls.push(
+        field.type.charAt(0).toUpperCase() +
+          field.type.slice(1) +
+          ' ' +
+          field.name
+      );
+    } else {
+      if (field.referenceTo.length > 1) {
+        decls.push('SObject' + ' ' + field.relationshipName);
+      } else {
+        decls.push(field.referenceTo + ' ' + field.relationshipName);
+      }
+      // field.type will be "reference", but the actual type is an Id for Apex
+      decls.push('Id ' + field.name);
+    }
+    return decls;
+  }
+
+  private static fieldName(decl: string) {
+    return decl.substr(decl.indexOf(' ') + 1);
+  }
+
+  // VisibleForTesting
+  public async generateFauxClass(
+    folderPath: string,
     sobject: any
-  ): Promise<void> {
-    const folderPath = path.join(projectPath, '.sfdx', 'sobjects');
+  ): Promise<string> {
     if (!fs.existsSync(folderPath)) {
-      //fs.mkdirSync(folderPath);
+      fs.mkdirSync(folderPath);
     }
-    //const isCustom = sobject.custom;
+    const fauxClassPath = path.join(folderPath, sobject.name + '.cls');
+
+    fs.writeFileSync(fauxClassPath, this.generateFauxClassText(sobject));
+
+    return fauxClassPath;
+  }
+
+  private generateFauxClassDecls(sobject: SObject): string[] {
+    const declarations: string[] = [];
+    if (sobject.fields) {
+      for (const field of sobject.fields) {
+        const decls: string[] = this.generateField(field);
+        if (decls && decls.length > 0) {
+          for (const decl of decls) {
+            declarations.push(decl);
+          }
+        }
+      }
+    }
+
+    if (sobject.childRelationships) {
+      for (const rel of sobject.childRelationships) {
+        if (rel.relationshipName) {
+          const decl: string = this.generateChildRelationship(rel);
+          if (decl) {
+            declarations.push(decl);
+          }
+        }
+      }
+      for (const rel of sobject.childRelationships) {
+        // handle the odd childRelationships last (without relationshipName)
+        if (!rel.relationshipName) {
+          const decl: string = this.generateChildRelationship(rel);
+          if (decl) {
+            declarations.push(decl);
+          }
+        }
+      }
+    }
+    return declarations;
+  }
+
+  private generateFauxClassTextFromDecls(
+    className: string,
+    declarations: string[]
+  ): string {
+    // sort, but filter out duplicates
+    // which can happen due to childRelationships w/o a relationshipName
+    declarations.sort(function nameCompare(
```

Comment:
Can we use the lambda syntax instead of declaring a function explicitly in there?